### DISCUSSION TITLE
Questions on 'calcGrains'

### DISCUSSION THREAD

**Discussion by bjzhou96** [mtex-toolbox/mtex#1293] [Ask Anything] [answered]
I am going to plot a phase map with subgrain boundary. I am folloing the document from https://mtextoolbox.github.io/SubGrainBoundaries.html. 
code:
[grains,ebsd.grainId] = calcGrains(ebsd);
ebsd(grains(grains.grainSize<12)) = [];
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',[1*degree,10*degree]);

% plot the ebsd data
plot(ebsd('Amphibole'),'faceAlpha',0.5,'figSize','large')

% init override mode
hold on

% plot grain boundares
plot(grains('Amphibole').boundary,'linewidth',0.5)

% % compute transparency from misorientation angle

% plot the subgrain boundaries
plot(grains.innerBoundary,'linewidth',2,'linecolor','b');

% stop override mode
hold off

However, an error came out: Index exceeds array bounds.

Error in EBSD/calcGrains>doSegmentation (line 151)
      quaternion(ebsd.rotations),ebsd.CSList{p},Dl(ndx),Dr(ndx),gbcValue(p),varargin{:});

Error in EBSD/calcGrains (line 33)
[A_Db,I_DG] = doSegmentation(I_FD,ebsd,varargin{:});

Error in subgrainboundary (line 39)
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',[1*degree,10*degree]);

The error is caused by two threshold in calcGrains. Is there anyone have the same trouble? When I set only one threshold:[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',10*degree);
it works. However, the map I got has much less subgrain boundary than that processed by Channel 5(subgrain boundary with misoritation between 2 and 10). I can't figure out what happened. 
When I set only one threshold, is that mean the boundars with misorientations between 0 and 10 are innerBoundary(subgrain boundary)? If so, it can be expected to obtain a map with more subgrain boundary than that from Channel 5.
Map from Mtex:[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',10*degree);
![image](https://user-images.githubusercontent.com/58097480/144529106-02191491-026d-4b8b-8659-d51cd625c7cd.png)

Map from Channel 5 : yellow boundary: 2-7; red boundary: 7-10
![PHASE](https://user-images.githubusercontent.com/58097480/144529347-196927dd-a414-42b4-a8ef-335a9de342b2.jpg)

How to plot more subgrain boundary? And is there any other way to plot a map with subgrain boundary?

**Comment by kilir** [comment, marked as answer]
Hi,
a) with respect to your first error: Do you use a recent version of mtex?

b) In mtex, everything which satisfies a grain boundary condition, but cannot be connected to a closed loop (or against the boundary of the map) is stored in innerBoundary. By default, this is a threshold angle above which a grain boundary is identified. Hence,using only one threshold angle and inspecting innerBoundary, mtex return all boundaries which a) satisfy the angle condition (e.g. > 10°) but b) do not form closed loops. You can check this by colorcoding your boundaries by the angle. This is not what you want to compare.

Hope that helps,
Rüdiger

**Reply by bjzhou96** [reply]
Hello Rüdiger
Thank you very much for your reply. Actually, I have tried with Mtex5.6.0., but it still didn't work. Thank you very much for your explanation for the innerboundary! Now I understand what it means. 

Thank you again for your help!

Best wishes

Baojun Zhou

**Reply by bjzhou96** [reply]
Hello Rüdiger
Sorry, I made a stupid mistake. I did download Mtex5.7.0., but I forgot to change the work path in Matlab. What I used is still the old version. Now I tried with the new one, and it works! Thank you very much for helping me. The question is finnally resovled! 
Best wishes

Baojun Zhou
